oqs-template/oqs-sig-info.md

```diff
@@ -14,16 +14,26 @@
 | dilithium3_aes **hybrid with** p384               | NIST Round 3 submission                       | 3            |                    3 | 0xfeab       | 1.3.9999.2.11.3          |
 | dilithium5_aes                                    | NIST Round 3 submission                       | 3            |                    5 | 0xfeac       | 1.3.6.1.4.1.2.267.11.8.7 |
 | dilithium5_aes **hybrid with** p521               | NIST Round 3 submission                       | 3            |                    5 | 0xfead       | 1.3.9999.2.11.4          |
-| falcon512                                         | 20211101                                      | 3            |                    1 | 0xfeae       | 1.3.9999.3.6             |
-| falcon512 **hybrid with** p256                    | 20211101                                      | 3            |                    1 | 0xfeaf       | 1.3.9999.3.7             |
-| falcon512 **hybrid with** rsa3072                 | 20211101                                      | 3            |                    1 | 0xfeb0       | 1.3.9999.3.8             |
+| falcon512                                         | 20211101                                      | 3            |                    1 | 0xfed7       | 1.3.9999.3.11            |
+| falcon512 **hybrid with** p256                    | 20211101                                      | 3            |                    1 | 0xfed8       | 1.3.9999.3.12            |
+| falcon512 **hybrid with** rsa3072                 | 20211101                                      | 3            |                    1 | 0xfed9       | 1.3.9999.3.13            |
+| falcon512                                         | PQClean Round 3 version                       | 3            |                    1 | 0xfeae       | 1.3.9999.3.6             |
```

Comment:
This seems fishy: Why do OIDs 1.3.999.3.6-8 change algorithm version identifier (from "20211101" to "PQClean Round 3 version"), but neither OIDs, nor code points? Or asked the other way around: Why does the same algorithm version "20211101" get different OIDs?

Reply:
I did it this way because due to [an old PQClean bug](https://github.com/open-quantum-safe/liboqs/issues/1561#issuecomment-1797883620), the "20211101" version of Falcon in PQClean / liboqs / oqs-provider doesn't actually implement the upstream 20211101-version spec; the long KATs don't match. The new updates are to correctly implement (hopefully) the 20211101 version of the spec.

I suppose (?) that we shouldn't retroactively re-version things, though... what do you think would be the best way to resolve this?

Reply:
Ouch. No, retroactively re-versioning doesn't sound like a good approach. I do now understand the reference to PQClean, though. But this shows that we have an extraordinarily weak and brittle approach to versioning. Created https://github.com/open-quantum-safe/liboqs/issues/1719 to track/improve. For this issue, what about adding to the old OIDs the (admittedly longish) term "PQClean Round 3 version labelled 20211101" ? That way, it becomes clear to the reader of this file that the same version ID had been used but had a different functionality.